src/components/TopicSelector.jsx

```diff
@@ -0,0 +1,41 @@
+import { getTopics } from "../api";
+import { useEffect, useState } from "react";
+import titleCase from "../utils/title-case";
+import { useNavigate } from "react-router-dom";
+
+const TopicSelector = ({ topic }) => {
+  console.log(topic);
+  const [topics, setTopics] = useState([]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    getTopics().then(({ topics }) => setTopics(topics));
+  }, []);
+
+  const redirect = (topic) => {
+    if (topic === "All Topics") {
+      navigate("/");
+    } else {
+      navigate(`/topics/${topic}`);
+    }
+  };
+
+  return (
+    <select
+      className="topic-selector"
+      onChange={(e) => redirect(e.target.value)}
+      value={topic ? topic : "All Topics"}
```

Comment:
You could possibly set the state of topic to be 'all topics' as its initial state - to get around this ternary